sdk/textanalytics/azure-ai-textanalytics/src/test/java/com/azure/ai/textanalytics/TextAnalyticsClientTestBase.java

```diff
@@ -0,0 +1,428 @@
+                                 DocumentResultCollection<T> expectedResult, String testApi) {
+        // assert batch result
+        assertEquals(expectedResult.getModelVersion(), actualResult.getModelVersion());
+        if (this.showStatistics) {
+            validateBatchStatistics(expectedResult.getStatistics(), actualResult.getStatistics());
+        }
+        validateDocuments(expectedResult, actualResult, testApi);
+
+        // TODO (savaity): refactor error model in azure-sdk-for-java#6559
+        // DocumentError error = new DocumentError().setId("4").setError("error");
+        // List<DocumentError> errors = new ArrayList<>();
+        // errors.add(error);
+        // validateErrorDocuments(expected.getErrors(), detectLanguageResultList);
+    }
+
+    /**
+     * Helper method to verify documents returned in a batch request.
+     *
+     * @param expectedResult the expected result collection..
+     * @param actualResult the actual result collection returned by the API.
+     * @param testApi the API to test.
+     */
+    private <T> void validateDocuments(DocumentResultCollection<T> expectedResult,
+                                       DocumentResultCollection<T> actualResult, String testApi) {
+        switch (testApi) {
+            case "Language":
+                final List<DetectLanguageResult> expectedResultList = expectedResult.stream()
+                    .filter(element -> element instanceof DetectLanguageResult)
+                    .map(element -> (DetectLanguageResult) element)
+                    .collect(Collectors.toList());
+
+                final List<DetectLanguageResult> actualResultList = actualResult.stream()
+                    .filter(element -> element instanceof DetectLanguageResult)
+                    .map(element -> (DetectLanguageResult) element)
+                    .collect(Collectors.toList());
+                assertEquals(expectedResultList.size(), actualResultList.size());
+
+                actualResultList.forEach(actualItem -> {
+                    Optional<DetectLanguageResult> optionalExpectedItem = expectedResultList.stream().filter(
+                        expectedEachItem -> actualItem.getId().equals(expectedEachItem.getId())).findFirst();
+                    assertTrue(optionalExpectedItem.isPresent());
+                    DetectLanguageResult expectedItem = optionalExpectedItem.get();
+                    if (actualItem.getError() == null && this.showStatistics) {
+                        validatePrimaryLanguage(expectedItem.getPrimaryLanguage(), actualItem.getPrimaryLanguage());
+                        validateDocumentStatistics(expectedItem.getStatistics(), actualItem.getStatistics());
+                        validateDetectedLanguages(expectedItem.getDetectedLanguages(), actualItem.getDetectedLanguages());
+                    }
+                });
+                break;
+            default:
+                break;
```

Comment:
Default, you should throw an exception rather than passing successfully. If I put testAPI, "language", the test case is written wrong and i should throw.